### ISSUE TITLE
`appush` & `appull`

### ISSUE THREAD

**Issue by parmentelat** [parmentelat/apssh#19]

**Comment by parmentelat** [comment]
second iteration about specifying the syntax for denoting local and remote files with these 2 yet-to-be-written commands

### how to denote remote files

the syntax uses
* `@:` as a way to replace the remote hostname when denoting remote files
* and optionnally formats like `{special}` to be replaced by the actual target's details, where the known specials are
  * `{user}` : username
  * `{fqdn}` : fdqn (as provided in each target)
  * `{host}`: same without the domain name (like hostname -s)

### pushing

copy the same local file over to a variety of hosts

- copy one local file on all remote home dirs

```
appush -t the_targets local_file @:
```
- likewise but with several local files, and store remote copies in /etc/

```
appush -t the_targets local_file1 local_file2 @:/etc
```

- in this third example here we would copy
  - `local-box1` onto `box1.inria.fr:/etc/some-file`
  - `local-box2` into `box2:/etc/some-file`

```
appush -t box1.inria.fr,box2 local-{host} @:/etc/some-file
```


### pulling

the same logic is at work for the other way around

- copy all the `/etc/fedora-release` files, into subdir `the-releases`, named as `<hostname>-fedora-release`

```
appull -t targets @:/etc/fedora-release the-releases/{host}-fedora-release
```

- same but fetch several files in one go; for each target the 2 files are stored under local `the-releases/box1.inria.fr` and `the-releases/box2`

```
appull -t -t box1.inria.fr,box2 @:/etc/fedora-release @:/etc/lsb-release the-releases/{fqdn}/
```

**Comment by parmentelat** [comment]
this post has been edited in anticipation for #20, so as to use the new naming {host} and {fqdn} instead

**Comment by parmentelat** [comment]
available in 0.24.0
minimally tested for now though
